README.md
# Fap Instructor

![Release](https://github.com/fapinstructor/fapinstructor-client/workflows/Release/badge.svg)


A fap/joi game where you follow the provided instructions while watching a stream of content that teases and torments you.

## Installation

### Client

```bash
cd client
yarn
yarn start
```

### Server

1. Configure your `.env` file under server. Use `.env.example` as a reference
2. Compile and run

```bash
cd server
yarn
yarn start
```

## Usage

[Adding a New Task](NEW_TASK.md)

## Contributing

Pull requests are welcome. For major changes, please open an issue first to discuss what you would like to change.

Please make sure to update tests as appropriate.


public/CHANGELOG.md
# Changelog

All notable changes to this project will be documented here.

The overall project roadmap and progress can be viewed on our [Trello Board](https://trello.com/b/JAIcnXRf/fap-instructor)

## 2023-03-26

- Fixed grip undefined notification bug.
- Fixed image height so scroll bars don't appear.
- Make the starting stroke speed slower.

## 2022-01-09

- Added local file system selector (only available in supported browsers)

## 2021-11-25

- Added support for handy stroke zones.

## 2021-10-22

- Fixed an issue where the game config wasn't being submitted.

## 2021-10-20

- Fixed handy devices that run v3 firmware.
- Fixed the unique subreddit validation errors.

## 2021-10-16

- Allow pasting multiple subreddits.

## 2021-09-30

- Complete overhaul of the config page.

## 2021-07-24

- Added redgifs and reddit gallery support
- Fixed and rewrote the final game logic (orgasm, denial, ruined)

## 2021-07-14

- Added Handy hardware support!
- Added new partner banner

## 2021-06-23

- Added button to view previous image
- Added hotkeys to the previous and next image buttons
- A ton of performance optimizations
- A variety of bug fixes

## 2021-05-05

- Added hotkeys to allow triggering actions via keyboard

## 2021-05-02

- Added ability to sort columns

## 2021-04-30

- Added ⭐ stars ⭐. You can now favourite any game and view them within the games table!
- Added "Average Game Length" column to game search

## 2021-04-14

- Added the ability for users to delete their profile

## 2021-02-22

- Added an FAQ page

## 2021-02-21

- Fixed the hands-off loop bug

## 2021-01-27

- Added the ability to hide the beat meter

## 2021-01-07

- Made more adjustments to a variety of tasks for smoother transitions
- Made grip adjustments occur more often and fixed a couple of bugs around them

## 2020-12-30

- Added metronome bar and ability to toggle sound
- Polish on a variety of tasks

## 2020-12-12

- Fixed issue where gifs weren't loading properly

## 2020-12-04

- Center images vertically for a better viewing experience
- Add prompt for when leaving an active game

## 2020-10-01

- Bookmarks now link back to the original post where the media came from
- Load times should be a thousand times faster
- Images are now preloaded to make slide transitions smoother
- The Game End page now has a nav bar
- Filtering videos, images, gifs should work flawlessly
- Duplicate images will not appear nearly as often because of a more expansive media library I pull from

## 2020-08-13

- Bug fixes
- Updated remaining packages

## 2020-08-10

- Bug fixes
- Updated packages

## 2020-08-04

- Fixed a bug for users who are unable to log in
- Lots of polish and code cleanup

## 2020-06-23

- Bug fixes
- Added a warmup period before the game kicks off

## 2020-06-22

- Performance improvements
- Some bug fixes
- Better bug tracking and monitoring

## 2020-06-09

- Added the ability to search for games
- Moved the previously played games and created games from the profile page into the games search page

## 2020-05-27

- Added a created games table under the profile page

## 2020-05-26

- Added a configure game option to every shared link
- Games can be made private or public when being shared. If they are public they will be searchable in the upcoming finder feature.
- Polish and tech debt

## 2020-05-22

- Polish and bug fixes
- Added "Play Game" button on Share Game dialog

## 2020-05-19

- Add a new share game button
- Add a tagging system
- Show previously played games on the profile page

## 2020-02-14

- Add a fullscreen button to the game page

## 2020-02-13

- Made Navbar responsive for mobile devices

## 2020-02-10

- Site navigation
- Change log
- Login Options
- Profile Page (Features to come)
- Add error message when failing to generate a game link

## 2020-01-17

- New game link generation. This will reduce the length of the URL making it easier to share.


src/audio.js
import { getRandomInclusiveInteger } from "utils/math";
import config from "config";

const { audioUrl } = config;

export const audioVariations = {
  Moan: 24,
  Faster: 2,
  Denied: 1,
  Orgasm: 3,
  Ruined: 3,
  CEI: 3,
};

export const getRandomAudioVariation = (key) => {
  if (!audioVariations.hasOwnProperty(key)) {
    throw new Error(`No audio variation found for ${key}`);
  }
  return audioLibrary[
    `${key}${getRandomInclusiveInteger(1, audioVariations[key])}`
  ];
};

const audioLibrary = {
  // cei
  CEI1: `${audioUrl}/cei/1.mp3`,
  CEI2: `${audioUrl}/cei/2.mp3`,
  CEI3: `${audioUrl}/cei/3.mp3`,
  // denied
  Denied1: `${audioUrl}/denied/1.mp3`,
  // faster
  Faster1: `${audioUrl}/faster/1.mp3`,
  Faster2: `${audioUrl}/faster/2.mp3`,
  // moan
  Moan1: `${audioUrl}/moan/1.mp3`,
  Moan2: `${audioUrl}/moan/2.mp3`,
  Moan3: `${audioUrl}/moan/3.mp3`,
  Moan4: `${audioUrl}/moan/4.mp3`,
  Moan5: `${audioUrl}/moan/5.mp3`,
  Moan6: `${audioUrl}/moan/6.mp3`,
  Moan7: `${audioUrl}/moan/7.mp3`,
  Moan8: `${audioUrl}/moan/8.mp3`,
  Moan9: `${audioUrl}/moan/9.mp3`,
  Moan10: `${audioUrl}/moan/10.mp3`,
  Moan11: `${audioUrl}/moan/11.mp3`,
  Moan12: `${audioUrl}/moan/12.mp3`,
  Moan13: `${audioUrl}/moan/13.mp3`,
  Moan14: `${audioUrl}/moan/14.mp3`,
  Moan15: `${audioUrl}/moan/15.mp3`,
  Moan16: `${audioUrl}/moan/16.mp3`,
  Moan17: `${audioUrl}/moan/17.mp3`,
  Moan18: `${audioUrl}/moan/18.mp3`,
  Moan19: `${audioUrl}/moan/19.mp3`,
  Moan20: `${audioUrl}/moan/20.mp3`,
  Moan21: `${audioUrl}/moan/21.mp3`,
  Moan22: `${audioUrl}/moan/22.mp3`,
  Moan23: `${audioUrl}/moan/23.mp3`,
  Moan24: `${audioUrl}/moan/24.mp3`,
  // orgas`m
  Orgasm1: `${audioUrl}/orgasm/1.mp3`,
  Orgasm2: `${audioUrl}/orgasm/2.mp3`,
  Orgasm3: `${audioUrl}/orgasm/3.mp3`,
  // ruined
  Ruined1: `${audioUrl}/ruined/1.mp3`,
  Ruined2: `${audioUrl}/ruined/2.mp3`,
  Ruined3: `${audioUrl}/ruined/3.mp3`,
  // misc.
  CardShuffle: `${audioUrl}/cardshuffle.mp3`,
  DontCumYet: `${audioUrl}/dontcumyet.mp3`,
  Edge: `${audioUrl}/edge.mp3`,
  HoldEdge: `${audioUrl}/holdedge.mp3`,
  KeepStroking: `${audioUrl}/keep_stroking.mp3`,
  LongMoan: `${audioUrl}/longMoan.mp3`,
  Obey: `${audioUrl}/obey.mp3`,
  RuinItForMe: `${audioUrl}/ruinitforme.mp3`,
  SlapBalls: `${audioUrl}/slapballs.mp3`,
  SqueezeBalls: `${audioUrl}/squeezeballs.mp3`,
  StartStrokingAgain: `${audioUrl}/start_stroking_again.mp3`,
  StartGame: `${audioUrl}/startgame.mp3`,
  ThereYouGo: `${audioUrl}/thereYouGo.mp3`,
  Tick: `${audioUrl}/tick.mp3`,
  Tighter: `${audioUrl}/tighter.mp3`,
};

export default audioLibrary;
